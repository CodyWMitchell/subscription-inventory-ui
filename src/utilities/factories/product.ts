import Faker from 'faker';
import { Factory } from 'fishery';
import { Product } from '../../hooks/useProducts';

export default Factory.define<Product>(() => ({
  name: Faker.vehicle.model(),
  productLine: Faker.vehicle.vehicle(),
  quantity: Faker.datatype.number(),
  sku: Faker.datatype.uuid(),
  serviceLevel: Faker.datatype.string(),
  serviceType: Faker.datatype.string(),
  unitOfMeasure: uOMFake
}));

const uOMFake = {
  name: Faker.datatype.string(),
  quantity: Faker.datatype.string()
};
