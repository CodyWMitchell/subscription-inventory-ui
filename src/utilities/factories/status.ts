import Faker from 'faker';
import { Factory } from 'fishery';
import { StatusCard } from '../../hooks/useStatus';

export default Factory.define<StatusCard>(() => ({
  active: Faker.datatype.number(),
  expiringSoon: Faker.datatype.number(),
  expired: Faker.datatype.number(),
  futureDated: Faker.datatype.number()
}));
