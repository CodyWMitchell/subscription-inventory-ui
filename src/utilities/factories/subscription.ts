import { faker } from '@faker-js/faker';
import { Factory } from 'fishery';
import { Subscription } from '../../hooks/useProducts';

export default Factory.define<Subscription>(() => ({
  quantity: faker.datatype.string(),
  number: faker.random.numeric(3),
  contractNumber: faker.random.numeric(3),
  status: faker.datatype.string(),
  endDate: faker.datatype.string(),
  startDate: faker.datatype.string()
}));
