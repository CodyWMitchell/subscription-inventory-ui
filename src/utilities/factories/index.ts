import product from './product';
import status from './status';

export default { product, status };
