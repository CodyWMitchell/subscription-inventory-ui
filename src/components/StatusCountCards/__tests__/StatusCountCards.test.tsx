import React, { FunctionComponent } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { render } from '@testing-library/react';
import { get, def } from 'bdd-lazy-var';
import factories from '../../../utilities/factories';
import StatusCountCard, { StatusCardProps } from '../StatusCountCards';

const queryClient = new QueryClient();

const Cards: FunctionComponent<StatusCardProps> = ({ statusCardData, statusIsFetching }) => (
  <QueryClientProvider client={queryClient}>
    <StatusCountCard statusCardData={statusCardData} statusIsFetching={statusIsFetching} />
  </QueryClientProvider>
);

describe('StatusCountCard', () => {
  def('loading', () => false);
  def('error', () => false);
  def('fetching', () => false);
  def('data', () => [
    factories.status.build({
      active: 10,
      expiringSoon: 3,
      expired: 1,
      futureDated: 2
    })
  ]);

  it('renders correctly', () => {
    const { container } = render(
      <Cards statusCardData={get('data')} statusIsFetching={get('fetching')} />
    );

    expect(container).toMatchSnapshot();
  });
});
