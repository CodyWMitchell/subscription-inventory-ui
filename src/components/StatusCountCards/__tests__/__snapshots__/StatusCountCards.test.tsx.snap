// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCountCard renders correctly 1`] = `
<div>
  <div
    class="pf-l-grid pf-m-gutter"
  >
    <div
      class="pf-l-grid__item"
    >
      <div
        class="pf-l-gallery pf-m-gutter"
        style="--pf-l-gallery--GridTemplateColumns--min: 350px;"
      >
        <div
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__body"
          >
            Active
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-inline-flex"
            >
              <div
                class="pf-l-flex pf-m-space-items-sm"
              >
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    fill="var(--pf-global--success-color--100)"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </div>
                <div
                  class=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-2"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__body"
          >
            Expiring soon
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-inline-flex"
            >
              <div
                class="pf-l-flex pf-m-space-items-sm"
              >
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    fill="var(--pf-global--warning-color--100)"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 576 512"
                    width="1em"
                  >
                    <path
                      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    />
                  </svg>
                </div>
                <div
                  class=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-3"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__body"
          >
            Expired
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-inline-flex"
            >
              <div
                class="pf-l-flex pf-m-space-items-sm"
              >
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    class="pf-u-danger-color-100"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    />
                  </svg>
                </div>
                <div
                  class=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-4"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe="true"
          id=""
        >
          <div
            class="pf-c-card__body"
          >
            Future dated
          </div>
          <div
            class="pf-c-card__body"
          >
            <div
              class="pf-l-flex pf-m-inline-flex"
            >
              <div
                class="pf-l-flex pf-m-space-items-sm"
              >
                <div
                  class=""
                >
                  <svg
                    aria-hidden="true"
                    fill="var(--pf-global--success-color--100)"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </div>
                <div
                  class=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
